pkg/controller/lifecycle/actuator.go

```diff
@@ -270,69 +320,13 @@ func (a *actuator) InjectScheme(scheme *runtime.Scheme) error {
 	return nil
 }
 
-func (a *actuator) getOrCreateTLSSecret(ctx context.Context, certificateConfig secrets.CertificateSecretConfig, namespace string) (*corev1.Secret, error) {
-	caSecret, ca, err := secrets.LoadCAFromSecret(ctx, a.client, namespace, v1beta1constants.SecretNameCACluster)
-	if err != nil {
-		return nil, fmt.Errorf("error fetching CA secret %s/%s: %v", namespace, v1beta1constants.SecretNameCACluster, err)
-	}
-
-	var (
-		secret = corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Annotations: make(map[string]string),
-				Name:        certificateConfig.Name,
-				Namespace:   namespace,
-			},
-		}
-		key = types.NamespacedName{
-			Name:      certificateConfig.Name,
-			Namespace: namespace,
-		}
-	)
-	if err := a.client.Get(ctx, key, &secret); client.IgnoreNotFound(err) != nil {
-		return nil, fmt.Errorf("error preparing kubeconfig: %v", err)
-	}
-
-	var (
-		computedChecksum   = utils.ComputeChecksum(caSecret.Data)
-		storedChecksum, ok = secret.Annotations[util.CAChecksumAnnotation]
-	)
-	if ok && computedChecksum == storedChecksum {
-		return &secret, nil
-	}
-
-	certificateConfig.SigningCA = ca
-	certificateConfig.CertType = secrets.ServerCert
-
-	config := secrets.ControlPlaneSecretConfig{
-		CertificateSecretConfig: &certificateConfig,
-		Name:                    certificateConfig.Name,
-	}
-
-	controlPlane, err := config.GenerateControlPlane()
-	if err != nil {
-		return nil, fmt.Errorf("error creating kubeconfig: %v", err)
-	}
-
-	_, err = controllerutil.CreateOrUpdate(ctx, a.client, &secret, func() error {
-		secret.Data = controlPlane.SecretData()
-		if secret.Annotations == nil {
-			secret.Annotations = make(map[string]string)
-		}
-		secret.Annotations[util.CAChecksumAnnotation] = computedChecksum
-		return nil
-	})
-
-	return &secret, err
-}
-
 func getLabels() map[string]string {
 	return map[string]string{
 		"app.kubernetes.io/name": constants.ApplicationName,
 	}
 }
 
-func getSeedResources(oidcReplicas *int32, hibernated bool, namespace, genericKubeconfigName string, shootAccessSecret *gutil.ShootAccessSecret) (map[string][]byte, error) {
+func getSeedResources(oidcReplicas *int32, hibernated bool, namespace, genericKubeconfigName, shootAccessSecretName, serverTLSSecretName string) (map[string][]byte, error) {
```

Comment:
In this function we should probably pass the new `bundle.crt` and use it in the kubeconfig that the apiserver uses to talk to the OWA here
https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/31612df0845d00e3ca1bb252885b080db81288fe/pkg/controller/lifecycle/actuator.go#L345

Reply:
OK, got it, thanks!